src/kOS/Binding/MissionSettings.cs

```diff
@@ -60,36 +60,24 @@ public override void AddTo(SharedObjects shared)
 
             shared.BindingMgr.AddGetter("TARGET", () =>
             {
-                if (shared.Vessel != FlightGlobals.ActiveVessel)
-                {
-                    throw new kOS.Safe.Exceptions.KOSSituationallyInvalidException("TARGET can only be returned for the Active Vessel");
-                }
-                var currentTarget = FlightGlobals.fetch.VesselTarget;
+                var target = shared.Vessel == FlightGlobals.ActiveVessel ?
+                    FlightGlobals.fetch.VesselTarget : shared.Vessel.targetObject;
 
-                var vessel = currentTarget as Vessel;
-                if (vessel != null)
-                {
+                if (target is Vessel vessel)
                     return VesselTarget.CreateOrGetExisting(vessel, shared);
-                }
-                var body = currentTarget as CelestialBody;
-                if (body != null)
-                {
+                if (target is CelestialBody body)
                     return BodyTarget.CreateOrGetExisting(body, shared);
-                }
-                var dockingNode = currentTarget as ModuleDockingNode;
-                if (dockingNode != null)
-                {
+                if (target is ModuleDockingNode dockingNode)
```

Comment:
How is it that I've been writing C# code for 12 years and I've never seen this syntax?  I admit that entire time has been as a hobyist, but it sure seems like a very useful feature.  I'm a little curious how it compares on performance in terms of allocations, but if it is essentially the same or better I'll gladly stash this away in my toolbox

Reply:
I've never seen this either.  C# is full of weird obfuscations like this that are IMO just there to be different, not because they're truly useful.  Never in a million years would I look at that line of code and guess what it does if not for seeing the old lines that it replaces in this diff output.  There's nothing in this syntax that visually hints at what it does.  You'd never guess that this is an assignment statement that puts a value into ``dockingNode`` with syntax like this.  You'd definitely be able to guess that there's a cast or type check happening, but not that there's any assignment occurring.

Reply:
It is one of the newer features added since open-sourcing Roslyn, C#6 or 7, not sure.
They have created many shortcuts I really like (e.g. expression body: `prop => value`),
less typing. Visual Studio is even suggesting such syntax.